server/app/services/openapi/AbstractOpenApiSchemaGenerator.java

```diff
@@ -28,4 +47,159 @@ protected ImmutableList<Scheme> getSchemes() {
+      Optional<Format> swaggerFormat,
+      DefinitionType arrayItemDefinitionType) {
+
+    if (definitionType == DefinitionType.ARRAY) {
+      return new ArrayProperty()
+          .items(
+              arrayItemDefinitionType == DefinitionType.STRING
+                  ? new StringProperty()
+                  : new ObjectProperty());
+    }
+
+    var property = new UntypedProperty();
+
+    property.setType(definitionType.toString());
+
+    if (swaggerFormat.isPresent()) {
+      property.setFormat(swaggerFormat.get().toString());
+    }
+
+    return property;
+  }
+
+  /** Get the list of scalars for a question sorted alphabetically */
+  protected static ImmutableList<Scalar> getScalarsSortedByName(
+      QuestionDefinition questionDefinition)
+      throws InvalidQuestionTypeException, UnsupportedQuestionTypeException {
+    return Scalar.getScalars(questionDefinition.getQuestionType()).stream()
+        .filter(scalar -> !excludeScalarFromSchemaOutput(scalar))
+        .map(AbstractOpenApiSchemaGenerator::remapScalar)
+        .sorted(Comparator.comparing(Enum::name))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  /** Get the field name from the scalar */
+  protected static String getFieldNameFromScalar(Scalar scalar) {
+    // Over output property name where needed
+    return switch (scalar) {
+      case CURRENCY_CENTS -> ApiPathSegment.CURRENCY_DOLLARS.toString().toLowerCase(Locale.ROOT);
+      case FILE_KEY_LIST -> ApiPathSegment.FILE_URLS.toString().toLowerCase(Locale.ROOT);
+      case NAME_SUFFIX -> ApiPathSegment.SUFFIX.toString().toLowerCase(Locale.ROOT);
+      default -> scalar.name().toLowerCase(Locale.ROOT);
+    };
+  }
+
+  /** Map Scalar to DefinitionType */
+  protected static DefinitionType getDefinitionTypeFromSwaggerType(Scalar scalar) {
+    // Override type because we messed up
+    if (scalar == Scalar.LATITUDE || scalar == Scalar.LONGITUDE || scalar == Scalar.WELL_KNOWN_ID) {
+      return DefinitionType.STRING;
+    }
+
+    switch (scalar.toScalarType()) {
+      case LONG:
+        return DefinitionType.INTEGER;
+      case CURRENCY_CENTS:
+      case DOUBLE:
+        return DefinitionType.NUMBER;
+      case LIST_OF_STRINGS:
+        return DefinitionType.ARRAY;
+      case DATE:
+      case STRING:
+      case PHONE_NUMBER:
+      case SERVICE_AREA:
+      default:
+        return DefinitionType.STRING;
+    }
+  }
+
+  /** Map ScalarType to Format */
+  protected static Optional<Format> getSwaggerFormat(Scalar scalar) {
+    // Override type because we messed up
+    if (scalar == Scalar.LATITUDE || scalar == Scalar.LONGITUDE || scalar == Scalar.WELL_KNOWN_ID) {
+      return Optional.empty();
+    }
+
+    switch (scalar.toScalarType()) {
+      case DATE:
+        return Optional.of(Format.DATE);
+      case LONG:
+        return Optional.of(Format.INT64);
+      case CURRENCY_CENTS:
+      case DOUBLE:
+        return Optional.of(Format.DOUBLE);
```

Comment:
do we need to be more careful with currency conversion then this? could we end up with a double like $90.010000000000001?